crates/router/src/core/webhooks/outgoing_v2.rs

```diff
@@ -0,0 +1,817 @@
+                        delivery_attempt,
+                        status_code.as_u16(),
+                        "Ignoring error when sending webhook to merchant",
+                        ScheduleWebhookRetry::NoSchedule,
+                    )
+                    .await?;
+                }
+            }
+        },
+    }
+
+    Ok(())
+}
+
+async fn raise_webhooks_analytics_event(
+    state: SessionState,
+    trigger_webhook_result: CustomResult<(), errors::WebhooksFlowError>,
+    content: Option<api::OutgoingWebhookContent>,
+    merchant_id: common_utils::id_type::MerchantId,
+    event: domain::Event,
+    merchant_key_store: &domain::MerchantKeyStore,
+) {
+    let key_manager_state: &KeyManagerState = &(&state).into();
+    let event_id = event.event_id;
+
+    let error = if let Err(error) = trigger_webhook_result {
+        logger::error!(?error, "Failed to send webhook to merchant");
+
+        serde_json::to_value(error.current_context())
+            .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
+            .inspect_err(|error| {
+                logger::error!(?error, "Failed to serialize outgoing webhook error as JSON");
+            })
+            .ok()
+    } else {
+        None
+    };
+
+    let outgoing_webhook_event_content = content
+        .as_ref()
+        .and_then(api::OutgoingWebhookContent::get_outgoing_webhook_event_content)
+        .or_else(|| get_outgoing_webhook_event_content_from_event_metadata(event.metadata));
+
+    // Fetch updated_event from db
+    let updated_event = state
+        .store
+        .find_event_by_merchant_id_event_id(
+            key_manager_state,
+            &merchant_id,
+            &event_id,
+            merchant_key_store,
+        )
+        .await
+        .attach_printable_lazy(|| format!("event not found for id: {}", &event_id))
+        .map_err(|error| {
+            logger::error!(?error);
+            error
+        })
+        .ok();
+
+    // Get status_code from webhook response
+    let status_code = updated_event.and_then(|updated_event| {
+        let webhook_response: Option<OutgoingWebhookResponseContent> =
+            updated_event.response.and_then(|res| {
+                res.peek()
+                    .parse_struct("OutgoingWebhookResponseContent")
+                    .map_err(|error| {
+                        logger::error!(?error, "Error deserializing webhook response");
+                        error
+                    })
+                    .ok()
+            });
+        webhook_response.and_then(|res| res.status_code)
+    });
+
+    let webhook_event = OutgoingWebhookEvent::new(
+        merchant_id,
+        event_id,
+        event.event_type,
+        outgoing_webhook_event_content,
+        error,
+        event.initial_attempt_id,
+        status_code,
+        event.delivery_attempt,
+    );
+    state.event_handler().log_event(&webhook_event);
+}
+
+fn get_webhook_url_from_business_profile(
```

Comment:
this should be me moved to impl function on profile, avoid using business_profile term instead usin profile